functions.php

```diff
@@ -32,15 +45,15 @@ function connect_db_return_collection($collectionName)
 
 function eachCharacter(array $character): string
 {
+    if (!isset($character['character_name']) || !isset($character['IQ']) || !isset($character['name']) || !isset($character['image'])) {
+        return '';
```

Comment:
not sure if the return works here but could be wrong . another thing is you could change this if statement to be 
`if (isset() &&.....)`
and then it'll only add to the $characterHtmlElementsString if those keys and values exists.